x-pack/platform/plugins/shared/fleet/server/services/agent_policy_create.ts

```diff
@@ -205,13 +222,11 @@ export async function createAgentPolicyWithPackages({
     try {
       await agentlessAgentService.createAgentlessAgent(esClient, soClient, agentPolicy);
     } catch (err) {
-      if (err instanceof AgentlessAgentCreateOverProvisionnedError) {
```

Comment:
Callout! I removed this conditional here, as it seems to be if we are creating an agentless agent policy, but fail to actually deploy it we should delete the policy so its not orphaned. I can't think of a case where we would want this to still stick around.